### PR TITLE
Migrate CollectivesWithData.js and CollectiveCard.js to styled components

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/3289

### CODE REVIEW

**Comment by Betree** [comment]
The easiest is to visit https://staging.opencollective.com/hosts

File: components/CollectivesWithData.js [hunk 1]
```diff
@@ -73,52 +74,28 @@ class CollectivesWithData extends React.Component {
...
+          display="flex"
+          flexWrap="wrap"
+          flexDirection="row"
+          justifyContent="center"
+          overflow="hidden"
+          margin="1rem 0"
+        >
           {collectives.map(collective => (
-            <CollectiveCard key={collective.id} collective={collective} />
+            <CollectiveCard style={{ margin: '1rem' }} key={collective.id} collective={collective} />
```
Comment (Betree) [inline components/CollectivesWithData.js:87]: I'm not sure that `CollectiveCard` accepts this prop. Could be nice to migrate `CollectiveCard` in the same PR as both components are closely tied.

Comment (SudharakaP) [reply components/CollectivesWithData.js]: I've corrected that and also migrated CollectiveCard in this PR. Feel free to let me know if anything further needs to be done. :smile:

**Review by Betree — APPROVED** [review]
Works like a charm! Thank you for doing this @SudharakaP, the work you're doing here is super helpful :muscle:

**Comment by SudharakaP** [comment]
> Works like a charm! Thank you for doing this @SudharakaP, the work you're doing here is super helpful muscle

You are welcome. Glad I could help. 😄

**Pull request merged**